pkg/kubelet/gpu/nvidia/nvidia_gpu_manager_test.go

```diff
@@ -60,6 +62,34 @@ func makeTestPod(numContainers, gpusPerContainer int) *v1.Pod {
 	return pod
 }
 
+func TestNewNvidiaGPUManager(t *testing.T) {
+	podLister := &testActivePodsLister{}
+
+	// Expects nil GPUManager and an error with nil dockerClient.
+	testGpuManager1, err := NewNvidiaGPUManager(podLister, nil)
+	as := assert.New(t)
+	as.Nil(testGpuManager1)
+	as.NotNil(err)
+
+	// Expects a GPUManager to be created with non-nil dockerClient.
+	fakeDocker := libdocker.NewFakeDockerClient()
+	testGpuManager2, err := NewNvidiaGPUManager(podLister, fakeDocker)
+	as.NotNil(testGpuManager2)
+	as.Nil(err)
+
+	// Expects zero capacity without any GPUs.
+	gpuCapacity := testGpuManager2.Capacity()
+	as.Equal(len(gpuCapacity), 1)
+	rgpu := gpuCapacity[v1.ResourceNvidiaGPU]
+	as.Equal(rgpu.Value(), int64(0))
+
+	err2 := testGpuManager2.Start()
+	if err2 != nil {
+		gpus := reflect.ValueOf(testGpuManager2).Elem().FieldByName("allGPUs").Len()
```

Comment:
Can you put a note of how many we do expect as a comment? Also will testGpuManager2.Start try to go through the discovery logic or do we want to hardcode it more?